securemailbox/views/retrieve.py

```diff
@@ -1,11 +1,69 @@
-from flask import Blueprint, jsonify
+from flask import Blueprint, jsonify, request, json
+
+from ..models import Mailbox
+from ..models import Message
 
 from securemailbox import db
 
 # Create the `retrieve` blueprint
 retrieve_blueprint = Blueprint("retrieve", __name__)
 
-
 @retrieve_blueprint.route("/retrieve/", methods=["POST"])
 def retrieve():
-    return jsonify({"success": True, "error": None})
+
+    data = request.get_json(force=True, silent=True)
+
+    if data is None:
+        return jsonify({'success': False, 'error': 'Request must be valid json', 'data': None}), 400
+
+    # Check if fingerprint is present in json request body
+    if 'fingerprint' not in data:
+        return jsonify({'success': False, 'error': 'fingerprint not in request', 'data': None}), 400
+
+    # Check if fingerprint value is string variable before query
+    if(isinstance(data['fingerprint'], str) == False):
```

Comment:
Nit: The "pythonic" way of writing this would be:
```suggestion
    if not isinstance(data['fingerprint'], str):
```

Noting that python doesn't require parens around conditions and implicitly checking type.